src/librustc_codegen_llvm/base.rs

```diff
@@ -192,12 +193,17 @@ pub fn compile_codegen_unit(
                                 bb_idx.index()
                             ))
                             .unwrap();
-                            let mut bx = Builder::with_cx(&cx);
-                            bx.position_at_end(llbb);
-                            bx.add_yk_block_label(llbb, lbl_name);
+                            labels.push((llbb, lbl_name));
                         }
                     }
                 }
+                // Only apply labels after collecting them in the previous step to avoid a double
+                // borrow of sir_cx
```

Comment:
Add a full-stop.

Reply:
And if we want to be totally precise, the problem is a double *mutable* borrow. Multiple immutable borrows are allowed.